tests/Metadata/Resource/Factory/FileLoaderResourceNameCollectionFactoryTest.php

```diff
@@ -0,0 +1,94 @@
+use ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\ConfigDummy;
+
+use ApiPlatform\Core\Metadata\Resource\Factory\XmlResourceMetadataFactory;
+use ApiPlatform\Core\Metadata\Resource\Factory\YamlResourceMetadataFactory;
+use ApiPlatform\Core\Metadata\Resource\ConfigurationMap;
+
+class FileLoaderResourceNameCollectionFactoryTest extends \PHPUnit_Framework_TestCase
+{
+    public function testHasResourceYaml()
+    {
+        $configPath = __DIR__.'/../../../Fixtures/TestBundle/Resources/config/resources.yml';
+        $configurationMap = new ConfigurationMap();
+        $ymlResourceNameCollectionFactory = new YamlResourceNameCollectionFactory([$configPath], $configurationMap);
+
+        $this->assertEquals($ymlResourceNameCollectionFactory->create(), new ResourceNameCollection([
+            ConfigDummy::class
+        ]));
+
+        $this->assertEquals($configurationMap->get(ConfigDummy::class), [
+                'shortName' => 'thedummyshortname',
+                'description' => 'Dummy resource',
+                'class' => 'ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\ConfigDummy',
+                'itemOperations' => ['my_op_name' => [], 'my_other_op_name' => []],
+                'collectionOperations' => ['my_collection_op' => []],
+                'iri' => 'someirischema'
+        ]);
+    }
+
+    public function testHasResourceXml()
+    {
+        
+        $configPath = __DIR__.'/../../../Fixtures/TestBundle/Resources/config/resources.xml';
+        $configurationMap = new ConfigurationMap();
+        $xmlResourceNameCollectionFactory = new XmlResourceNameCollectionFactory([$configPath], $configurationMap);
+
+        $this->assertEquals($xmlResourceNameCollectionFactory->create(), new ResourceNameCollection([
+            ConfigDummy::class
+        ]));
+
+        $this->assertEquals($configurationMap->get(ConfigDummy::class), [
+                'shortName' => 'thedummyshortname',
+                'description' => 'Dummy resource',
+                'class' => 'ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\ConfigDummy',
+                'itemOperations' => ['my_op_name' => [], 'my_other_op_name' => []],
+                'collectionOperations' => ['my_collection_op' => []],
+                'iri' => 'someirischema'
+        ]);
+    }
+
+    public function testYamlCreateResourceMetadata()
+    {
+        $configPath = __DIR__.'/../../../Fixtures/TestBundle/Resources/config/resources.yml';
+        $configurationMap = new ConfigurationMap();
+        $yamlResourceNameCollectionFactory = new YamlResourceNameCollectionFactory([$configPath], $configurationMap);
+        $yamlResourceNameCollectionFactory->create();
+
+        $resourceMetadataFactory = new YamlResourceMetadataFactory($configurationMap);
+        $resourceMetadata = $resourceMetadataFactory->create(ConfigDummy::class);
+
+        $this->assertInstanceOf(ResourceMetadata::class, $resourceMetadata);
+    }
```

Comment:
@teohhanhui for example. 

Now I need to see if `createMetadata` should be kept in the `ResourceMetadataFactoryTrait`, which depends on the use of `Annotation/Resource` as a pre-metadata class (see https://github.com/api-platform/core/pull/490/commits/94634982474ac5bccef9b3e2e9d62778313d5323#r58549969).